packages/components/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  createRef,
+  useEffect,
+  useState,
+} from "react";
+import classnames from "classnames";
+import ReactDOM from "react-dom";
+import { AnimatePresence, motion } from "framer-motion";
+import { Text } from "../Text";
+import styles from "./Tooltip.css";
+
+interface TooltipProps {
+  readonly children: ReactElement;
+  /**
+   * Tooltip text
+   */
+  readonly message: string;
+}
+
+export function Tooltip({ message, children }: TooltipProps) {
+  const [direction, setDirection] = useState("above");
+  const [showOnLoad, setShowOnLoad] = useState(false);
+  const [position, setPosition] = useState({ top: "0px", left: "0px" });
+  const tooltipRef = createRef<HTMLDivElement>();
+  const shadowRef = createRef<HTMLSpanElement>();
+
+  showOnHover(shadowRef, setShowOnLoad);
+
+  useEffect(() => {
+    if (
+      tooltipRef.current &&
+      shadowRef.current &&
+      shadowRef.current.nextElementSibling
+    ) {
+      const activator = shadowRef.current.nextElementSibling;
+      const tooltip = tooltipRef.current;
+      const activatorBounds = activator.getBoundingClientRect();
+      const xOffset =
+        activatorBounds.right -
+        activatorBounds.width / 2 -
+        tooltip.clientWidth / 2;
+
+      if (activatorBounds.top <= 100) {
```

Comment:
Do you think we should make this check relative to the viewport?

Reply:
I don't think so. At least not in this case just because the tooltip just needs to avoid showing up outside of the viewport.. which means if the bounds goes negative, it's outside of the viewport